Source/Core/Core/PowerPC/Interpreter/Interpreter_Integer.cpp

```diff
@@ -356,21 +356,10 @@ void Interpreter::srawx(UGeckoInstruction inst)
   else
   {
     int amount = rb & 0x1f;
-    if (amount == 0)
-    {
-      rGPR[inst.RA] = rGPR[inst.RS];
-      SetCarry(0);
-    }
-    else
-    {
-      s32 rrs = rGPR[inst.RS];
-      rGPR[inst.RA] = rrs >> amount;
+    s32 rrs = rGPR[inst.RS];
+    rGPR[inst.RA] = rrs >> amount;
 
-      if ((rrs < 0) && (rrs << (32 - amount)))
-        SetCarry(1);
-      else
-        SetCarry(0);
-    }
+    SetCarry(rrs < 0 && (u32(rrs) << (32 - amount)) != 0);
```

Comment:
Is this correct for the `amount == 0` -> `SetCarry(0)` case from before? Same down below